web/src/app/components/timeline/timeline.component.ts

```diff
@@ -390,9 +398,20 @@ export class TimelineComponent implements AfterViewInit {
             .attr(
                 'style',
                 (d: Item) =>
-                    `fill: ${TimelineComponent.STATE_TO_COLOR[d.state]}`);
+                    `fill: ${TimelineComponent.STATE_TO_COLOR[d.state]}`)
+            .attr(
+                'stroke',
+                (d: Item) => {
+                    // If state color is light gray and has a duration less than
+                    // five minutes, set border to a dark gray for visibility.
+                    const color = TimelineComponent.STATE_TO_COLOR[d.state];
+                    const duration = d.endTime - d.startTime;
+                    return (color === '#eee' && duration < 300000) ?
```

Comment:
I think it would be good to make `#eee` and `300000` constants. They are also used 453.